File: Sean的leetcode/Leetcode/Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int min_len = INT_MAX;
        int sum = 0;
        for (int left = 0, right = 0; right < nums.size(); right++)
        {
            sum += nums[right];
            while (sum >= s)
            {
                min_len = min(min_len, right - left + 1);
                sum -= nums[left++];
            }
        }
        return min_len == INT_MAX ? 0 : min_len;
    }
};

File: Sean的leetcode/FB/Minimum Size Subarray Sum.py
###
# Two Pointers
# Time Complexity: O(n)
# Space Complexity: O(1)
###
class Solution(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        l , r = 0, 0
        tmp = 0
        min_len = float('inf')
        while r < len(nums):
            tmp += nums[r]
            while tmp >= s:
                min_len = min(min_len, r - l + 1)
                tmp -= nums[l]
                l += 1
            r += 1
        return min_len if min_len != float('inf') else 0
            